subsys/bluetooth/host/hci_core.c

```diff
@@ -746,13 +746,23 @@ static void le_conn_complete(struct net_buf *buf)
 		return;
 	}
 
-	id_addr = find_id_addr(&evt->peer_addr);
+	bt_addr_le_copy(&id_addr, &evt->peer_addr);
+
+	/* Translate "enhanced" identity address type to normal one */
+	if (id_addr.type == BT_ADDR_LE_PUBLIC_ID ||
+	    id_addr.type == BT_ADDR_LE_RANDOM_ID) {
+		id_addr.type -= BT_ADDR_LE_PUBLIC_ID;
+		bt_addr_copy(&peer_addr.a, &evt->peer_rpa);
```

Comment:
This is not always valid, should you check here?

Reply:
What do you mean? If id_addr.type is one of the tested values that causes this branch to be taken then evt->peer_rpa is required to be a valid RPA.

Reply:
Right sorry, I confused this with something else.